include/mxnet/operator.h

```diff
@@ -261,20 +261,28 @@ class OperatorProperty {
                           std::vector<int> *aux_type) const {
     CHECK_LE(in_type->size(), this->ListArguments().size());
     int n_in = this->ListArguments().size();
-    for (unsigned i = 0; i < in_type->size(); ++i) {
-      CHECK(in_type->at(i) == mshadow::default_type_flag ||
-            in_type->at(i) == -1) << "Unsupported data type " << in_type->at(i);
+    int dtype = -1;
```

Comment:
The default infertype forces float32 because most ops don't support other types. If your op does, override this in that op